### PR TITLE
Warning Fixies

### PR DESCRIPTION
**Pull request by ethteck**
Fixed most warnings in the project. There are some world ones that will be be easier to handle as clover works through the maps. The non-world ones that remain are either useful (since they indicate things that need to be handled correctly) or non-trivial to fix.

### CODE REVIEW

**Comment by BowserSlug** [comment]
✅ This PR fixes 307 warnings!

File: src/entity_model.c [hunk 1]
```diff
@@ -222,7 +222,7 @@ s32 step_entity_model_commandlist(EntityModel* entityModel) {
             imageData = entityModel->gfx.imageData;
             entityModel->nextFrameTime = *curPos++;
             imageData->raster = (u8*)*curPos++;
-            imageData->defaultPal = (u8*)*curPos++;
+            imageData->defaultPal = (u16*)*curPos++;
```
Comment (HailSanta) [inline src/entity_model.c:225]: PAL_PTR

File: src/npc.c [hunk 2]
```diff
@@ -1054,7 +1055,7 @@ void npc_reload_all(void) {
                     if (!(npc->flags & NPC_FLAG_1000000) && (npc->palSwapType != 0)) {
                         npc->spritePaletteList = spr_get_npc_palettes(npc->currentAnim >> 16);
                         npc->paletteCount = 0;
-                        while (npc->spritePaletteList[npc->paletteCount] != -1) {
+                        while (npc->spritePaletteList[npc->paletteCount] != (u16*) -1) {
```
Comment (HailSanta) [inline src/npc.c:1058]: PAL_PTR

File: include/common_structs.h [hunk 3]
```diff
@@ -127,9 +127,9 @@ typedef struct CamPosSettings {
 } CamPosSettings; // size = 0x1C
 
 typedef struct DmaTable {
-    /* 0x00 */ s32 start;
-    /* 0x04 */ s32 end;
-    /* 0x08 */ s32 dest;
+    /* 0x00 */ u8* start;
+    /* 0x04 */ u8* end;
+    /* 0x08 */ u8* dest;
```
Comment (HailSanta) [inline include/common_structs.h:132]: Addr for these three?

**Review by HailSanta — APPROVED** [review]

File: include/common_structs.h (hunk 3 above)
Comment (ethteck) [reply include/common_structs.h:132]: Addr is u8[] so it unfortunately can't be used here (syntax errors)

**Pull request merged**